loadfile/loadfile_test.go

```diff
@@ -0,0 +1,102 @@
+package loadfile_test
+
+import (
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/engine/loadfile"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// This tests the functional behavior of LoadContext when it is given
+// a directory of files that cover more than the directory file node
+// type 'd' (as seen in *nix systems). Specifically, this test adds
+// symbolic link files for a regular file and a directory file. The
+// engine should only attempt to read files of a type that the os can
+// read (i.e. not throw an error on a call to os.ReadFile()), and
+// disregard (not throw an error) files with a type it cannot read.
+func TestLoadContext(t *testing.T) {
+	testdir := "/tmp/loadfile-test"
+	// cleanup directory even if it's there
+	_ = os.RemoveAll(testdir)
+
+	assert.NoError(t, os.MkdirAll(testdir, os.ModePerm))
+
+	// create a directory and a file
+	dirname := "mydir"
+	dirpath := filepath.Join(testdir, dirname)
+	filename := "myfile"
+	assert.NoError(t, os.MkdirAll(dirpath, os.ModePerm))
+	f, err := os.CreateTemp(testdir, filename)
+	assert.NoError(t, err)
+	tempfilepath := f.Name()
+	assert.NoError(t, f.Close())
```

Comment:
It seems weird to intermingle the steps to create the file with those for creating the directory, given that they are independent of each other (so, swap lines 28 & 29).

It also seems weird to use `os.CreateTemp()`:  you don't need the random name (or, at least, the directory didn't need a random name, so it seems odd that the file would), so `os.Create()` seems like a better choice (it saves you from having to fetch the file name).

Reply:
I would still need to make some call somewhere to get the file's absolute path, either with `filepath.Join()` or `filepath.Abs()`.

Reply:
> I would still need to make some call somewhere to get the file's absolute path, either with filepath.Join() or filepath.Abs().

I see...so you're using `CreateTemp()` as a clever substitute for `Join()`....  I'm not sure that that's really a win:  given that creating the directory is done in four steps (assign `dirname`, construct `dirpath`, call `os.MkdirAll()`, and check the result), having the file creation done in an analogous four-step process (assign `filename`, construct the file path, call `os.Create()`, and check the result) is very easy for the reader to apprehend.

Nice touch on adding the `Close()` call, btw.  🙂